p8e-util/src/main/kotlin/io/p8e/crypto/SmartKeySigner.kt

```diff
@@ -84,41 +83,14 @@ class SmartKeySigner(
     }
 
     override fun update(data: ByteArray, off: Int, res: Int) {
-        // If off is less then res, these are the data that we care about.
-        if(off < res) {
-            if(!verifying) {
-                val dataSample = data.copyOfRange(off, off+res)
-                signatureRequest?.data = dataSample
-            } else {
-                /**
-                 * The downstream (data verification) chunks the data into data size of 8192.
-                 * The data needs to be aggregated to its signing size of 32768 before the
-                 * data can be validated.
-                 */
-                if(objSizeIndexer == OBJECT_SIZE_BYTES) {
-                    objSizeIndexer = (off + res)
-                    aggregatedData = null
-                    aggregatedData = if (aggregatedData == null) {
-                        data.copyOfRange(off, off + res)
-                    } else {
-                        aggregatedData?.plus(data.copyOfRange(off, off + res))
-                    }
-                } else {
-                    objSizeIndexer += (off + res)
-                    aggregatedData = aggregatedData?.plus(data.copyOfRange(off, off + res))
-                }
-            }
+        if(!verifying) {
+            signatureRequest?.data = data.copyOfRange(off, res)
```

Comment:
couldn't this update method be called multiple times in the case of a stream of data... in that case, shouldn't the signatureRequest's data be appended to, otherwise this would just overwrite the data that we will then send to SmartKey for signing, right?

Reply:
No, for the signing case, the data comes in one big blob to get signed, i have an example of what the data kind of looks like (its huge). It's the verify that comes in smaller broken up chunks, hence the previous appended logic to handle the case.